### PR TITLE
Add pagination to Tableau Metadata API

### PR DESCRIPTION
**Pull request by mars-lan**
### 🤔 Why?

Tableau's [Metadata API](https://help.tableau.com/current/api/metadata_api/en-us/index.html) places a [limit](https://help.tableau.com/current/api/metadata_api/en-us/docs/meta_api_errors.html#:~:text=might%20be%20incomplete.-,NODE_LIMIT_EXCEEDED,-Warning) on the number of nodes returned. When exceeded, it'll return partial results, which can lead to incomplete lineage.

An example of the corresponding GraphQL error: 

```
    "errors": [
        {
            "message": "Showing partial results. The request exceeded the 20000 node limit. Use pagination, additional filtering, or both in the query to adjust results.",
            "extensions": {
                "severity": "WARNING",
                "code": "NODE_LIMIT_EXCEEDED",
                "properties": {
                    "nodeLimit": 20000
                }
            }
        }
    ]
```

### 🤓 What?

Add [pagination](https://help.tableau.com/current/api/metadata_api/en-us/docs/meta_api_examples.html#pagination) to the custom SQL tables & workbook GraphQL queries.

### 🧪 Tested?

Verified against a production instance.

### CODE REVIEW

**Review by alyiwang — COMMENTED** [review]
Otherwise, LGTM
great work!

File: metaphor/tableau/graphql_utils.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    result: List[Dict] = []
+
+    while True:
+        logger.info(f"Querying {connection_name} with offset {offset}")
+        resp = server.metadata.query(query, {"first": batch_size, "offset": offset})
+        if resp.get("errors"):
+            logger.error(f"Error when querying {connection_name}: {resp.get('errors')}")
+
+        nodes = resp["data"][connection_name]["nodes"]
+        if len(nodes) == 0:
```
[the remark this answers is not in the document]
Comment (mars-lan) [reply metaphor/tableau/graphql_utils.py]: Makes sense.

File: metaphor/tableau/graphql_utils.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    while True:
+        logger.info(f"Querying {connection_name} with offset {offset}")
+        resp = server.metadata.query(query, {"first": batch_size, "offset": offset})
+        if resp.get("errors"):
+            logger.error(f"Error when querying {connection_name}: {resp.get('errors')}")
+
+        nodes = resp["data"][connection_name]["nodes"]
+        if len(nodes) < batch_size:
+            return result
```
Comment (alyiwang) [inline metaphor/tableau/graphql_utils.py:26]: this should be after `result.extend()`

Comment (mars-lan) [reply metaphor/tableau/graphql_utils.py:28]: Thanks for catching. Updated tests to cover this.

**Review by alyiwang — APPROVED** [review]

**Pull request merged**